recipes/spdlog/all/conanfile.py

```diff
@@ -109,28 +126,27 @@ def package(self):
             tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
             tools.rmdir(os.path.join(self.package_folder, "lib", "spdlog", "cmake"))
 
-    def package_id(self):
-        if self.options.header_only:
-            self.info.header_only()
-
     def package_info(self):
-        if self.options.header_only:
-            component_name = "spdlog_header_only"
-        else:
-            component_name = "libspdlog"
```

Comment:
This is a breaking change. consumers could have components requiring spdlog::spdlog_header_only.

Reply:
Yes it's a breaking change, conditional component name is bad. This component will even disapear from this recipe in conan v2, it was a side effect of `.names` mechanism.